src/impl_methods.rs

```diff
@@ -2444,6 +2444,46 @@ where
         }
     }
 
+    /// Remove the `index`th elements along `axis` and shift down elements from higher indexes.
+    ///
+    /// Decreases the length of `axis` by one.
+    ///
+    /// ***Panics** if `axis` or `index` is out of bounds.
+    pub fn shift_remove_index(&mut self, axis: Axis, index: usize)
+    where
+        S: DataOwned + DataMut,
+    {
+        // TODO: It's possible to choose to shift the elment to the front or the back here,
+        // whichever is closer.
+        let (_, mut tail) = self.view_mut().split_at(axis, index);
```

Comment:
I think we should also check that `index < self.len_of(axis)`. The `split_at` method allows an `index` equal to the length of the axis, in which case the upper portion of the split has zero length. I don't think it makes sense to allow this case, because the index wouldn't really be *removed* -- it wasn't a valid index in the first place.